builtin/providers/triton/resource_firewall_rule_test.go

```diff
@@ -0,0 +1,152 @@
+package triton
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/joyent/gosdc/cloudapi"
+)
+
+func TestAccTritonFirewallRule_basic(t *testing.T) {
+	config := testAccTritonFirewallRule_basic
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckTritonFirewallRuleDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckTritonFirewallRuleExists("triton_firewall_rule.test"),
```

Comment:
Definitely not a dealbreaker, but noting that we're missing a `testCheckTritonFirewallRuleAttributes` func here, so we're not verifying that the attributes are making it up to the API, only that they make it into the state.